charts/s3gw/questions.yaml

```diff
@@ -1,73 +1,106 @@
 ---
 questions:
   # General settings
+  - variable: serviceName
+    default: s3gw
+    description: "S3 Service Name"
+    label: "S3 Service Name"
+    required: true
+    type: string
+    group: "General"
+
   - variable: accessKey
     default: test
     description: "S3 Access Key"
-    label: "Default Access Key"
+    label: "S3 Access Key"
     required: true
     type: string
     group: "General"
+
   - variable: secretKey
     default: test
     description: "S3 Secret Key"
-    label: "Default Secret Key"
+    label: "S3 Secret Key"
     required: true
     type: string
     group: "General"
+
   - variable: ingress.enabled
     default: true
     description: "Deploy an Ingress (Required for TLS and UI)"
     label: "Enable Ingress"
     required: true
     type: boolean
     group: "General"
-  - variable: hostname
+
+  - variable: publicDomain
     show_if: ingress.enabled=true
-    default: s3gw.local
-    description: "Hostname of the S3 Service (FQDN)"
-    label: "Hostname"
+    default: 127.0.0.1.omg.howdoi.website
+    description: "Public domain of the S3 Service used by the Ingress"
+    label: "Public Domain"
     required: true
     type: string
     group: "General"
+
+  - variable: privateDomain
+    default: svc.cluster.local
+    description: "Private domain of the S3 Service used inside the Kubernetes cluster"
+    label: "Private Domain"
+    required: true
+    type: string
+    group: "General"
+
   - variable: ui.enabled
     default: false
     description: "UI Enabled"
     label: "UI Enabled"
     required: true
     type: boolean
     group: "General"
-  - variable: ui.hostname
-    show_if: ui.enabled=true && ingress.enabled=true
-    default: s3gw-ui.local
-    description: "Hostname of the S3 UI (FQDN)"
-    label: "UI Hostname"
+
+  - variable: ui.serviceName
+    default: s3gw-ui
+    description: "UI Service Name"
+    label: "UI Service Name"
+    required: true
+    type: string
+    group: "General"
+
+  - variable: ui.publicDomain
+    show_if: ingress.enabled=true
+    default: 127.0.0.1.omg.howdoi.website
+    description: "Public domain of the UI Service used by the Ingress"
+    label: "UI Public Domain"
+    required: true
     type: string
     group: "General"
 
   # TLS settings
   - variable: tls.crt
     show_if: ingress.enabled=true
     description: |
-      TLS Certificate for the Gateway (base64 encoded, CN must match hostname)
+      TLS Certificate for the Gateway (base64 encoded, CN must match serviceName + publicDomain)
```

Comment:
It's probably best to describe the requirements for the certificates in our documentation and just link that here. I don't think we can describe all the details with wildcard certs etc. in one line accurate enough.
That being said, for now this seems ok to me.